load_balancer/test/integration/load_balancerTest.js

```diff
@@ -0,0 +1,243 @@
+    responseBody.components.forEach((elem, index) => {
+      if (upArray.includes(index)) {
+        elem.status.should.equal('up');
+      } else {
+        elem.status.should.equal('down');
+      }
+    });
+    restoreConsole();
+    resp();
+  });
+};
+
+const submitAssert = function submitAssert() {
+  const tempJob = testData.submissionsData[0];
+  request.post(`${lbUrl}/submit`, { json: tempJob }, (error, response) => {
+    (error === null).should.be.true();
+    response.body.should.equal(true);
+  });
+};
+
+describe('Correctly maintains list of ENs', () => {
+  beforeEach(() => {
+    stubConsole();
+    startLoadBalancer();
+    restoreConsole();
+    activateNocks();
+  });
+
+  afterEach((done) => {
+    cleanNocks();
+    loadBalancer.server.close(done);
+  });
+
+  it('during status check', (done) => {
+    stubConsole();
+    const res = JSON.parse(JSON.stringify(configData.Nodes));
+    const lbTestStatus = configData.load_balancer;
+    lbTestStatus.status = 'up';
+    [0, 1].forEach((elem) => {
+      connectionCheckNock(`https://${res[elem].hostname}:${res[elem].port}`, true);
+      connectionCheckNock(`https://${res[elem].hostname}:${res[elem].port}`, true);
+    });
+    request.get(`${lbUrl}/connectionCheck`, (error, response) => {
+      const responseBody = JSON.parse(response.body);
+      response.should.exist();
+      responseBody.should.have.all.keys('components', 'job_queue_length', 'timestamp');
+      responseBody.job_queue_length.should.equal(loadBalancer.job_queue.length);
+      responseBody.timestamp.should.be.a('string');
+      responseBody.components.should.be.an('array');
+      responseBody.components.should.deep.include(lbTestStatus);
+      responseBody.components.forEach((elem, index) => {
+        if (index === 0 || index === 1 || index === 10) {
+          elem.status.should.equal('up');
+        } else {
+          elem.status.should.equal('down');
+        }
+      });
+      restoreConsole();
+      done();
+    });
+  });
+});
+
+describe('Correctly adds a newly started node', () => {
+  beforeEach(() => {
+    stubConsole();
+    startLoadBalancer();
+    restoreConsole();
+    activateNocks();
+  });
+
+  afterEach((done) => {
+    cleanNocks();
+    loadBalancer.server.close(done);
+  });
+
+  it('when no job is pending', (done) => {
+    const tempNode = testData.nodesData[0];
+    stubConsole();
+    connectionCheckNock(`https://${tempNode.hostname}:${tempNode.port}`, false);
+    connectionCheckNock(`https://${tempNode.hostname}:${tempNode.port}`, true);
+    const requestRunNockObj = requestRunNock(`https://${tempNode.hostname}:${tempNode.port}`);
+    request.post(`${lbUrl}/addNode`, { json: tempNode });
+    setTimeout(() => {
+      requestRunNockObj.isDone().should.equal(false);
+      connectionCheckAssert([0, 10], () => {
+        done();
+      });
+    }, 50);
+  });
+
+  it('when a job is pending', (done) => {
+    const tempNode = testData.nodesData[0];
```

Comment:
This test is similar to the previous test. You can have a function wrap the code that exists between lines 150 to 162. The function can return boolean value which can be tested in the `ìt codeblock`.

It is quite common to have functions which will be specific to tests. You just found one of them. Use this technique for all the code repetitions.

Reply:
Please wrap the code in a function. The differences can be treated as different conditional blocks inside the function. Apart from the required arguements, the function can take a boolean variable named `queueEmpty`. When the queue is empty, submit() need not be called.
You can return the true/false value from the function so that the unit test can proceed forward.